### PR TITLE
Linter: remove dead code and address unhandled errors

### PR DESCRIPTION
**Pull request by kalensk**
This pulls in @berlin-ab linter work to 1) remove dead code, and 2) address unhandled errors exposed by golangci-lint.

We are breaking the linter work into digestible chunks ideally grouping similar issues together into a single commit. This will keep the PRs short and specific to make it easy to review, and to get this merged before the code gets stale and drifts. Further PR's to follow.

[Test Pipeline](https://cm.ci.gpdb.pivotal.io/teams/main/pipelines/gpupgrade:fix_linter_issues_part1)

### CODE REVIEW

File: hub/upgrade_standby_test.go [hunk 1]
```diff
@@ -19,7 +19,10 @@ func TestUpgradeStandby(t *testing.T) {
 		}
 
 		runner := newSpyRunner()
-		hub.UpgradeStandby(runner, config)
+		err := hub.UpgradeStandby(runner, config)
+		if err != nil {
+			t.Errorf("unexpected error: %#v", err)
```
Comment (jchampio) [inline hub/upgrade_standby_test.go:24]: nit: `%+v` for unexpected errors.

File: step/stream_test.go [hunk 2]
```diff
@@ -61,8 +61,14 @@ func TestMultiplexedStream(t *testing.T) {
...
 		for i := 0; i < 10; i++ {
-			stream.Stdout().Write([]byte{'O'})
-			stream.Stderr().Write([]byte{'E'})
+			_, err := stream.Stdout().Write([]byte{'O'})
+			if err != nil {
+				t.Errorf("unexpected error: %#v", err)
+			}
+			_, err = stream.Stderr().Write([]byte{'E'})
+			if err != nil {
+				t.Errorf("unexpected error: %#v", err)
```
Comment (jchampio) [inline step/stream_test.go:70]: Ditto. (And `unexpected error` isn't helpful context in this case; maybe `writing stdout:` and `writing stderr:`?)

Comment (bhuvnesh2703) [reply step/stream_test.go]: fixed

File: hub/upgrade_standby_test.go (hunk 1 above)
Comment (bhuvnesh2703) [reply hub/upgrade_standby_test.go]: fixed

**Pull request merged**